app/helpers/Emails/Notifications/Submissions/SubmissionEmailsSender.php

```diff
@@ -69,31 +70,45 @@ public function submissionEvaluated(AssignmentSolutionSubmission $submission): b
       return false;
     }
 
-    // Handle evaluation notification...
+    // Handle evaluation notifications...
+    return $this->handleEvaluationNotifications($solution, $submission, $user) &&
+      $this->handleTeacherNotifications($solution, $submission, $user);
```

Comment:
AFAIK the PHP interpret uses short-circuit evaluation. It means that teacher notifications will not be sent if the handleEvaluationNotifications() method yields false. Is this really desired behavior?

Reply:
Well what should I say, I am an idiot... of course this is not desired...